react-components/src/components/Architecture/SettingsButton.tsx

```diff
@@ -177,30 +186,33 @@ function createToggle(command: BaseCommand, t: TranslateDelegate): ReactNode {
 
   const label = command.getLabel(t);
   return (
-    <Menu.ItemAction
-      key={uniqueId}
-      label={label}
-      disabled={!isEnabled}
+    <StyledToggleContainer
       onClick={() => {
-        command.invoke();
-        setChecked(command.isChecked);
-      }}
-      trailingContent={<Switch checked={isChecked} disabled={!isEnabled} />}
-    />
+        if (isEnabled) {
+          command.invoke();
+        }
+      }}>
+      <Switch checked={isChecked} disabled={!isEnabled} />
+      <TextLabel text={label} />
+    </StyledToggleContainer>
```

Comment:
This seem to change a lot. We don't seem to do the `setChecked(command.isChecked)` call anymore. And I didn't see any tests around this as well 🤔 What do we do here really?